doc/userguide/devguide/codebase/installation-from-git.rst

```diff
@@ -0,0 +1,168 @@
+to ensure that you have everything you need for the installation.
+
+.. code-block:: bash
+
+  sudo apt-get -y install libpcre3 libpcre3-dbg libpcre3-dev \
+  build-essential autoconf automake libtool libpcap-dev libnet1-dev \
+  libyaml-0-2 libyaml-dev pkg-config zlib1g zlib1g-dev libcap-ng-dev \
+  libcap-ng0 make libmagic-dev libjansson-dev rustc cargo jq git-core
+
+Add ``${HOME}/.cargo/bin`` to your path.
+
+.. code-block:: bash
+
+  export PATH=$PATH:${HOME}/.cargo/bin
+  cargo install --force cbindgen
+
+Depending on the current status of your system, it may take a while to
+complete this process.
+
+**IPS**
+
+By default, Suricata works as an IDS. If you want to use it as a IDS and IPS
+program, enter:
+
+.. code-block:: bash
+
+  sudo apt-get -y install libnetfilter-queue-dev libnetfilter-queue1 \
+  libnfnetlink-dev libnfnetlink0
+
+Suricata
+~~~~~~~~
+
+First, it is convenient to create a directory for Suricata.
+Name it 'suricata' for example. Open the terminal and enter:
+
+.. code-block:: bash
+
+  mkdir suricata
+
+
+Followed by:
+
+.. code-block:: bash
+
+  cd suricata
+
+Next, enter the following line in the terminal:
+
+.. code-block:: bash
+
+  git clone git@github.com:OISF/suricata.git
+  cd suricata
+
+Libhtp is not bundled. Get libhtp by doing:
+
+.. code-block:: bash
+
+  ./scripts/bundle.sh libhtp
+
+Followed by:
+
+.. code-block:: bash
+
+  ./autogen.sh
+
+
+To configure, please enter:
+
+.. code-block:: bash
+
+  ./configure
+
+
+To compile, please enter:
+
+.. code-block:: bash
+
+  make
+
+To install Suricata, enter:
+
+.. code-block:: bash
+
+  sudo make install
+  sudo ldconfig
+
+To install suricata-update
+
+Follow the instructions found
+https://suricata-update.readthedocs.io/en/latest/quickstart.html
+
+.. note:: If you would like to build ``suricata-update`` from source, enter:
+
+  .. code-block:: bash
+
+    ./scripts/bundle.sh suricata-update
+    cd suricata-update
+    python setup.py build
+    sudo python setup.py install
+    sudo suricata-update
+
+Auto setup
+~~~~~~~~~~
+
+You can also use the available auto setup features of Suricata:
+
+
+ex:
+
+.. code-block:: bash
+
+  ./configure && make && sudo make install-conf
+
+*make install-conf*
+would do the regular "make install" and then it would automatically
+create/setup all the necessary directories and ``suricata.yaml`` for you.
+
+.. code-block:: bash
+
+  ./configure && make && make install-rules
+
+*make install-rules*
+would do the regular "make install" and then it would automatically download
+and set up the latest ruleset from Emerging Threats available for Suricata
+
+.. code-block:: bash
+
+  ./configure && make && make install-full
+
+*make install-full*
+would combine everything mentioned above (install-conf and install-rules) -
+and will present you with a ready to run (configured and set up) Suricata
+
+
+Please continue with
+`Basic Setup <https://redmine.openinfosecfoundation.org/projects/suricata/
+wiki/Basic_Setup>`_.
```

Comment:
Let's replace this link with the one we have on readthedocs:

https://suricata.readthedocs.io/en/latest/quickstart.html#basic-setup

feel free to create an explicit target for the Basic setup section in the quickstart file, so you can use `ref` :)

(If you are unfamiliar with explicit targets for Sphinx, I find this useful:  https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#explicit-targets)

Reply:
I had no idea we had `basic-setup` on the readthedocs already (I should've checked, my bad)

Thanks for the cross-referencing link. 👍🏿

Reply:
It's no problems, it's my duty to double-check these things ;)